C++/C.sublime-syntax

```diff
@@ -429,125 +482,312 @@ contexts:
+          scope: storage.modifier.c
+        - match: '{{identifier}}'
+          scope: entity.name.member.indexed.c debug.member-variable-identifier-list.function-pointer
+    - match: '{{common_identifier}}'
+      scope: entity.name.member.c debug.member-variable-identifier-list
+      comment: There are a lot of common variable names 4 characters or under.
+      push:
+        - meta_scope: debug.member-variable-identifier-list.array
+        - include: declaration-brackets
+        - include: or_pop
     - match: '{{identifier}}'
-      scope: entity.name.struct.c
-    - match: '(?=;)'
+      scope: entity.name.member.indexed.c debug.member-variable-identifier-list
+      push:
+        - meta_scope: debug.member-variable-identifier-list.array
+        - include: declaration-brackets
+        - include: or_pop
+    - match: ','
+      scope: punctuation.separator.c debug.member-variable-identifier-list
+    - match: ';'
+      scope: punctuation.terminator.c
       pop: true
+    - include: or_pop
+
+  struct-keyword:
+    - match: '\bstruct\b'
+      scope: storage.type.c
+      push:
+        - meta_scope: meta.struct.c debug.struct-keyword
+        - include: data-structures-definition-common-begin
+        - match: '{{identifier}}(?=\s*{|\s*\n)'
+          scope: entity.name.struct.c
+          comment: Struct definition
+          set: struct-body
+        - match: '{{identifier}}'
+          scope: support.type.c
+          comment: Struct declaration
+          pop: true
+        - match: '(?={)'
+          comment: Anonymous struct definition
+          set: struct-body
+        - include: or_pop
+  struct-body:
+    - meta_content_scope: debug.struct-body
+    - include: comments
     - match: '\{'
-      scope: meta.block.c punctuation.section.block.begin.c
+      scope: punctuation.section.block.begin.c
       set:
-        - meta_content_scope: meta.struct.c meta.block.c
+        - meta_scope: meta.struct.body.c meta.block.c debug.struct-body.block
+        - include: comments
         - match: '\}'
-          scope: meta.struct.c meta.block.c punctuation.section.block.end.c
+          scope: punctuation.section.block.end.c
           pop: true
         - include: data-structures-body
+    - include: or_pop
 
-  data-structures-enum-definition:
-    - meta_scope: meta.enum.c
-    - include: data-structures-definition-common-begin
-    - match: '{{identifier}}(?=\s*;)'
-      scope: entity.name.enum.forward-decl.c
-    - match: '{{identifier}}'
-      scope: entity.name.enum.c
-    - match: '(?=;)'
-      pop: true
+  enum-keyword:
+    - match: '\benum\b'
+      scope: storage.type.c
+      push:
+        - meta_scope: meta.enum.c debug.enum-keyword
+        - include: data-structures-definition-common-begin
+        - match: '{{identifier}}(?=\s*{|\s*\n)'
+          scope: entity.name.enum.c
+          comment: Struct definition
+          set: enum-body
+        - match: '{{identifier}}'
+          scope: support.type.c
```

Comment:
Like the other instances, is this the appropriate scope here?

Reply:
IMO yes, there is no real way to tell the difference between library/framework and user defined types.